### PR TITLE
Allow to configure service registries via helm

### PR DESCRIPTION
**Pull request by Nino-K**
Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[x] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Review by rshriram — APPROVED** [review]
nice!

File: install/kubernetes/helm/istio/charts/pilot/values.yaml [hunk 1]
```diff
@@ -52,3 +52,8 @@ podAntiAffinityTermLabelSelector: []
 # to a pilot. It balances out load across pilot instances at the cost of
 # increasing system churn.
 keepaliveMaxServerConnectionAge: 30m
+
+# list of platform service registries to read from
+# options are Counsul, Kubernetes, MCP, Mock
```
Comment (morvencao) [inline install/kubernetes/helm/istio/charts/pilot/values.yaml]: nit: s/Counsul/Consul/

**Comment by howardjohn** [comment]
When would a user set this? It seems to me like no one would ever set mock, Kubernetes is the default, and setting MCP does nothing (https://github.com/istio/istio/blob/d51f81dd1391283f1ae00aa2eabcfc8ac57f8225/pilot/pkg/bootstrap/server.go#L838-L839). Is this to turn off Kubernetes registry?

**Comment by Nino-K** [comment]
> When would a user set this? It seems to me like no one would ever set mock, Kubernetes is the default, and setting MCP does nothing (
> 
> https://github.com/istio/istio/blob/d51f81dd1391283f1ae00aa2eabcfc8ac57f8225/pilot/pkg/bootstrap/server.go#L838-L839
> 
> ). Is this to turn off Kubernetes registry?

@howardjohn yes, this is to disable k8's registry also I have a [PR](https://github.com/istio/istio/pull/12276) under review which enables MCP registry.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**